### ISSUE TITLE
crash application

### ISSUE THREAD

**Issue by pmaster555** [mfontanini/libtins#329]
```
SnifferConfig.set_promisc_mode(true);
SnifferConfig.set_immediate_mode(true);
SnifferConfig.set_filter("udp dst portrange 27000-30000");
NetworkInterface iface = NetworkInterface::default_interface();
Sniffer sniffer(iface.name(), SnifferConfig); // at this point the application crashes
```
The error occurs on approximately 5% of computers. Windows 7 or Windows 10, there is no difference. WinPcap installed at all the same. What could it be?

**Comment by pmaster555** [comment]
Unfortunately not. on the computer where I am doing the assembly I could not reproduce this error, it occurs in a very small number of people. 

But I noticed that if you write as written in the example, i.e.
http://libtins.github.io/tutorial/sniffing/windows
```
NetworkInterface iface = NetworkInterface::default_interface();
Sniffer sniffer(iface.name());

```
Without specifying the config, the application also crashes, so it should be?

**Comment by mfontanini** [comment]
That should not crash, the config is optional. Is the default interface's name valid before creating the sniffer? Also, are you saying something throws an exception or does it crash? Because if it throws an exception, the you should catch it and print the error message in it.

**Comment by mfontanini** [comment]
Regarding your UDP comment, that will throw an exception if the packet doesn't have a UDP layer. You seem to not be handling exceptions properly in your code.

**Comment by pmaster555** [comment]
> Это не должно произойти сбой, конфиг не является обязательным. Допустимо ли имя интерфейса по умолчанию перед созданием сниффера? Кроме того, вы говорите, что что-то выдает исключение или происходит сбой? Потому что, если оно выдает исключение, вы должны перехватить его и напечатать в нем сообщение об ошибке.

Yes, the interface name is normal, like {6527cc7d-c647-4986-ac10-7784dc1f2439}, 
I am not a professional programmer, so I can’t understand why, a small number of people fail during the sniffer initialization (
and unfortunately, I still do not know how to handle failures and exceptions on c ++
